bittensor/utils/networking.py

```diff
@@ -178,22 +163,49 @@ def get_formatted_ws_endpoint_url(endpoint_url: Optional[str]) -> Optional[str]:
 def ensure_connected(func):
     """Decorator ensuring the function executes with an active substrate connection."""
 
+    def is_connected(substrate) -> bool:
+        """Check if the substrate connection is active."""
+        sock = substrate.websocket.sock
+        return (
+            sock is not None
+            and sock.getsockopt(socket.SOL_SOCKET, socket.SO_ERROR) == 0
+        )
+
+    @retry(
+        exceptions=ConnectionRefusedError,
+        tries=5,
+        delay=5,
+        backoff=1,
+        logger=logging,
+    )
+    def reconnect_with_retries(self):
+        """Attempt to reconnect with retries using retry library."""
+        logging.info("Attempting to reconnect to substrate...")
+        self._get_substrate()
+
+        old_level = logging.get_level()
+        logging.set_info()
+        logging.success("Connection successfully restored!")
+        logging.setLevel(old_level)
```

Comment:
I don't understand why we need this.

Reply:
yeah, after https://github.com/opentensor/bittensor/pull/2446 we don't need this logic.
after we merge 2446 I gonna apply btlogging.console everywhere we need it (like in this case).